static/js/reactions.js

```diff
@@ -103,25 +103,10 @@ export function toggle_emoji_reaction(message_id, emoji_name) {
     // clicking on a reaction and that is handled by `process_reaction_click()`
     // method. This codepath is to be used only where there is no chance of an
     // user interacting with a deactivated realm emoji like emoji picker.
-    const reaction_info = {
-        emoji_name,
-    };
 
-    if (emoji.active_realm_emojis.has(emoji_name)) {
-        if (emoji_name === "zulip") {
-            reaction_info.reaction_type = "zulip_extra_emoji";
-        } else {
-            reaction_info.reaction_type = "realm_emoji";
-        }
-        reaction_info.emoji_code = emoji.active_realm_emojis.get(emoji_name).id;
-    } else {
-        const codepoint = emoji.get_emoji_codepoint(emoji_name);
-        if (codepoint === undefined) {
-            blueslip.warn("Bad emoji name: " + emoji_name);
-            return;
-        }
-        reaction_info.reaction_type = "unicode_emoji";
-        reaction_info.emoji_code = codepoint;
+    let reaction_info = {emoji_name, emoji_alt_code: page_params.emojiset === "text"};
+    if (!reaction_info.emoji_alt_code) {
```

Comment:
Why do we need this `emoji_alt_code` check?  The previous `reactions.js` code did not have that behavior.

The goal for a refactor like this should be to not change the existing logic for reactions, and it seems like it could be a bug to do so here.  It's only the new `user_status` logic that might want special handling for that emojiset, I think.